---
title: "Functional Enrichment Analysis of Sex-specific Muscle Transcriptome from Sedentary,
  Strength-Trained, or Endurance-Trained Participants"
author: "Erik Dassoff"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, comment = "", message = FALSE)

```

# Introduction

There is increasing interest in delivering personalized guidance to improve sports performance as well as overall health. One potential direction is to identify sex-specific differences in exercise adaptations to potentially optimize training regimens to improve health and performance goals. Of particular interest is identifying strategies to optimize cardiorespiratory fitness in different populations, given that cardiorespiratory fitness is increasingly recognized as a better predictor of all-cause mortality than levels of physical activity (Davidson et al. 2018). It is especially important to identify any potential sex differences, considering that women have historically been under-represented in exercise science research.

Considering that exercise results in widespread physiological adaptations, bioinformatics can be used as a starting point to identify potential sex-specific adaptations, by simultaneously assessing many pathways that are acutely or chronically up or down-regulated by exercise. Landen et al. (2019) previously reviewed many sex-specific alterations in muscle gene expression, DNA methylation, histone modification, and non-coding RNA expression. Some studies have pointed to differences in muscle proteolysis, oxidative phosphorylation, and tissue remodelling (Landen et al. 2019). These differences could lead to altered chronic adaptations. However, chronic adaptations have not been thoroughly investigated or are inconsistent.

The objectives of this project are to examine gene expression data to identify differentially expressed genes and functionally enriched pathways, comparing long-term well-trained participants with untrained controls in both males and females. Additionally, this project seeks to identify if any gene expression networks altered by endurance training differ between these groups. Throughout this process, this project will also help to learn novel tools used in functional enrichment analysis.

# Description of Data Set

Chapman et al. (2020) investigated sex-specific differences in the skeletal muscle transcriptome, comparing groups with over 15 years of endurance or strength-training history to untrained controls. This gene expression data contains a wealth of information on chronic adaptations from endurance training, including potential sex-specific differences. The dataset used in this project was obtained on 25 Nov 2023, as an excel file from supplementary differential gene expression (DGE) data (Chapman et al. 2020). The data contains information on gene names, p-values from differential expression analysis, fold change values, and direction of regulation. Prior steps conducted by Chapman et al. (2020) involved DGE analysis and quality control via the DESeq2 workflow. All groups are initially compared via histograms and volcano plots, to explore the entire available data set; however, enrichment analysis focuses specifically on addressing the question of sex-specific endurance training adaptations.

# Section 1: Data Acquisition, Exploration, Filtering, and Quality Control

```{r, Part1, fig.height=15}
#attributions: This script was adapted from vignettes for enhanced volcano plots (Blighe et al. 2018) and functional enrichment analysis via the pathfindR package (Ulgen et al. 2019). Additionally, data was obtained from Chapman et al. (2020) for differential gene expression analysis. Additional R tutorials were consulted throughout, as indicated via hyperlinks located in the script.----

#loading libraries----
library(readxl)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(data.table)
#BiocManager::install('EnhancedVolcano')
library(EnhancedVolcano)
library(gridExtra)
library(pathfindR)
library(microbenchmark)
library(utils)
library(cowplot)

#setting adjustable parameters----
#p-value for statistical significance, considering p-value already adjusted for false discovery rate. 0.05 is selected to consider a level of statistical significance that limits the false discovery rate of all significant results to less than 5%.
pCutoff <- 0.05

#cutoff for fold change for volcano plots. Two is selected to highlight reasonably large differentially expressed genes that are likely to have high biological relevance.
FCcutoff <- 2

#p-value cutoff for significant enrichment. 0.05 selected as a standard level and to not be overly exclusive.
enrichThresh <- 0.05

#adjustment method for enrichment analysis (i.e., bonferroni or fdr). Using fdr to be consistent with differential expression analysis.
adjMethod <- "fdr"

#number of iterations for protein sub-network search. Start with ten and increase for greater coverage of functional pathways.
iterations <- 10

#gene set for enrichment analysis (choose from: "Kegg", "Reactome", "BioCarta", "GO-all", "Go-BP", "GO-CC", "GO-MF", "cell_markers", "mmu_KEGG"). Starting with KEGG, because it contains biological pathways of interest with good interpretability.
geneSet <- "KEGG"

#setting the protein interaction network database to search to create gene interaction networks (choose from: "Biogrid", "STRING", "KEGG", "GeneMania", "IntAct", "mmu_STRING"). Starting with KEGG to be consistent with enrichment analysis choice.
pinName <- "KEGG"

#setting two algorithms to try for search optimization and to test the execution time (choose from greedy search ["GR"], simulated annealing ["SA"], and genetic algorithm ["GA"]). Starting with greedy search, since it is performs a truncated optimization to approximate active protein interaction network, but with less computational demands, since it isn't searching until finding a true optimum, like the SA method.
searchMethod <- "GR"

#input the terms that are desired to be compared. The code currently accepts two comparisons to controls, with the intent to further compare these differential gene expression profiles.
comparison1 <- "FEvsFC"
comparison2 <- "MEvsMC"

#loading dataset----
#Muscle tissue differential gene expression data from the following article as a downloadable excel file in the xlsx format: https://www.sciencedirect.com/science/article/pii/S2211124720307890#abs0015

#insert path to file
path <- "C:/Users/Erik/OneDrive - University of Guelph/Documents/BINF/enduranceTrainingDEG.xlsx"

#the data is being read into R with the readxl package. lapply is used to load every sheet in the excel file.
DEGlist <- lapply(excel_sheets(path), read_excel, path = path)

#check
class(DEGlist)

#finding sheet names. Help: https://readxl.tidyverse.org/reference/excel_sheets.html
sheetNames <- excel_sheets(path)

#renaming sheet names in list
names(DEGlist) <- sheetNames

#check
names(DEGlist)

#combining all comparisons into one data frame to make it easier to create multiple figures in one grid using facet wrap, and for other downstream processing.
allDEGcomparisons <- rbindlist(DEGlist, idcol = "comparison")

#checking that all group comparisons are in the new data frame
unique(allDEGcomparisons$comparison)
view(allDEGcomparisons)

#adding negative log10 p-values for downstream analysis
allDEGcomparisons$negLogP <- -log10(allDEGcomparisons$`P-Adjusted`)

#checking that column was added and that values are correct
names(allDEGcomparisons)
all.equal(allDEGcomparisons$negLogP, -log10(allDEGcomparisons$`P-Adjusted`))

#selecting variables to set up format for pathfindR downstream
selectForPathfindR <- function (x) {
  x %>%
    select(gene_name, Log2FoldChange, `P-Adjusted`)
}

DEGlistPathfindR <- lapply(DEGlist, selectForPathfindR)

#check that column names are as expected
lapply(DEGlistPathfindR, function(x) {names(x)})


#exploring data----

#creating function to explore data in every element of the list
explore <- function(x) {
  class = class(x)
  names = names(x)
  dimensions = dim(x)
  return(list(class = class, names = names, dimensions = dimensions))
}

lapply(DEGlist, explore)

#creating function to better understand the data values
dataStats <- function(x) {
  pValStats = summary(x$'P-Adjusted')
  FCstats = summary(x$'Log2FoldChange')
  numGeneNames = length(unique(x$'gene_name'))
  directions = table(x$'Direction')
  return(list("P Value Stats" = pValStats, "Fold Change Stats" = FCstats, "Number of Unique Genes" = numGeneNames, "Change Directions" = directions))
}

lapply(DEGlist, dataStats)

#checking to see if there are the same number of unique gene names as gene names overall for each list element. If so, these can serve as the unique identifiers for each element in each list.
lapply(DEGlist, function(x) {length(x$gene_name) == length(unique(x$gene_name))})

```

```{r, Quality, fig.height=4, fig.cap="Amount of significantly differentially regulated genes by p-value and group comparison. Abbreviations: F - Female; M - Male; E - Endurance; C - Control"}
#advice for quality checking for differential gene expression analysis: https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/08_practical_DE.pdf

#help with facetwrap:http://zevross.com/blog/2019/04/02/easy-multi-panel-plots-in-r-using-facet_wrap-and-facet_grid-from-ggplot2/

#plotting histogram of p-values for all comparisons with color to identify which are up- or down-regulated. Any comparisons with only up- or down-regulated genes are likely to not be of high quality.
GGhist <- ggplot(allDEGcomparisons, aes(`P-Adjusted`, fill = Direction))
GGhist +
  geom_histogram(binwidth = 0.001, col = I("grey10")) +
  labs(y = "Number of Differentially Expressed Genes", x = "Adjusted P-Value") +
  theme_classic() +
  facet_wrap(~allDEGcomparisons$comparison) +
  scale_fill_viridis_d(option = "D")

```

```{r, Volcano, fig.height=12, fig.cap="Volcano plots of differentially expressed genes for all combinations of male and female endurance athletes and their controls. Abbreviations: F - Female; M - Male; E - Endurance; C - Control; P - Significant (adj. P < 0.05) differential expression; FC_P - Significant differential expression (adj. P < 0.05) with fold change > 2", warning = FALSE}
#creating volcano plots to visualize significantly differentially expressed genes with high significance

#enhanced volcano plot vignette: https://bioconductor.org/packages/devel/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html

#creating a function for the volcano plot, to visualize significantly differentially expressed genes with high significance. Formatting is adjusted to fit in the Rmarkdown output.
volcanoPlot <- function(data, title) {EnhancedVolcano(data,
                lab = data$gene_name,
                x = 'Log2FoldChange',
                y = 'P-Adjusted',
                title = title,
                titleLabSize = 14,
                legendLabSize = 14,
                legendIconSize = 6,
                axisLabSize = 14,
                legendPosition = "top",
                pCutoff = pCutoff,
                FCcutoff = FCcutoff,
                pointSize = 3,
                labSize = 4,
                boxedLabels = TRUE,
                drawConnectors = TRUE,
                gridlines.minor = FALSE)}

#plotting result for each comparison.
p1 <- volcanoPlot(DEGlist$FEvsFC, "FE vs FC")
p2 <- volcanoPlot(DEGlist$MEvsMC, "ME vs MC")
p3 <- volcanoPlot(DEGlist$MCvsFC, "MC vs FC")
p4 <- volcanoPlot(DEGlist$MSvsME, "MS vs ME")
p5 <- volcanoPlot(DEGlist$MEvsFE, "ME vs FE")
p6 <- volcanoPlot(DEGlist$MSvsMC, "MS vs MC")

#change colours to viridis scale
viridis <- function(p) {
  p + scale_colour_viridis_d(option = "D")
}

p1 <- viridis(p1)
p2 <- viridis(p2)
p5 <- viridis(p5)
p3 <- viridis(p3)

#arranging selected figures into a grid
grid.arrange(p1, p2, p5, p3, ncol = 2)
```

# Main Software Tools

The main software tool used in this analysis was pathfindR (Ulgen et al. 2019). The associated vignette provided guidance on available features, options, and code formatting. This analysis extended on the vignette by writing new composite functions and selecting tailored settings for this analysis. This software was chosen, since it has been used in well-cited papers in the nutrition and exercise science fields and provides novel improvements to functional enrichment analysis. The approach differs from the more common over-representation analysis (ORA) by collectively accounting for the significance of differentially expressed genes by first mapping them onto optimized protein networks before enrichment analysis. This helps to better account for gene interactions and create more realistic representations of enriched pathways. I also found the tool to provide easily-interpretable visualizations and high flexibility to adjust the analysis to preferred settings. Examples of customizability include the ability to select algorithms used for identifying active gene subnetworks according to use-case and computational availability, the ability to access multiple types of protein interaction networks/gene sets for enrichment analysis, and the ability to adjust some plot features with GGplot code. Still, the additional optimization steps beyond ORA could present higher computational demands, especially for greater numbers of comparisons or longer gene lists.

# Section 2: Main Analysis

```{r, Enrichment}
#create functional annotation ---- using pathfindR: https://pubmed.ncbi.nlm.nih.gov/31608109/

#full tutorial: https://cran.r-project.org/web/packages/pathfindR/vignettes/intro_vignette.html

#converting to data frame for all comparisons, which is an essential class structure to be able to run downstream pathfindR analysis
DEGlistPathfindR <- lapply(DEGlistPathfindR, function (x) {as.data.frame(x)})

#check
lapply(DEGlistPathfindR, function (x) {class(x)})

#turning off scientific notation to match pathway analysis format
options(scipen = 999)

#check that values are not in scientific notation
summary(DEGlistPathfindR$FEvsFC$`P-Adjusted`)

#creating function for running pathfindR, which will be used more than once downstream. Not plotting the enrichment chart, so that it can be plotted separately later.
pathfindR <- function(data) {run_pathfindR(data,
  adj_method = adjMethod,
  gene_sets = geneSet,
  pin_name_path = pinName,
  search_method = "GS",
  enrichment_threshold = enrichThresh,
  plot_enrichment_chart = FALSE)}

#creating custom function for selecting key outputs of interest: clustering pathway enrichment with agglomerative hierarchical clustering and plotting the enriched terms as well as the gene interaction networks for each comparison.
enrichAnalysis <- function(data) {
  result <- run_pathfindR(data)
  clusteredEnrichment <- cluster_enriched_terms(result, plot_clusters_graph = FALSE)
  termGeneGraph <- term_gene_graph(result, use_description = TRUE)
  return(list("result" = result, "enrichment" = clusteredEnrichment, "termPlot" = termGeneGraph))
}

#creating pre-specified search list
df1 <- DEGlistPathfindR[[comparison1]]
df2 <- DEGlistPathfindR[[comparison2]]
inputList <- list(df1, df2)
remove(df1, df2)

#checking that the list contains two data frames
class(inputList)
length(inputList)
class(inputList[[1]])
view(inputList[[1]])

#running analysis function through pre-selected comparisons
combinedResult <- lapply(inputList, enrichAnalysis)

#viewing tabular results
view(combinedResult[[1]]$result)
view(combinedResult[[1]]$enrichment)

#creating a subset of samples with fold enrichment >2, low p-values, and only the representative terms for each cluster, to simplify figure by not having several terms within each cluster.
selectedClusters1 <- subset(combinedResult[[1]]$enrichment, Status %in% "Representative" & Fold_Enrichment > 1.5 & lowest_p < 0.01)

selectedClusters2 <- subset(combinedResult[[2]]$enrichment, Status %in% "Representative" & Fold_Enrichment > 1.5 & lowest_p <0.01)

#checking that subsetting was performed
length(combinedResult[[1]]$enrichment$ID) > length(selectedClusters1$ID)
```

```{r, results = FALSE, fig.height = 10, fig.cap="Functional enrichment analysis for female and male endurance athletes compared to their respective sedentary controls. All terms with enrichment p-values < 0.01 and fold change values above 1.5 are included. Abbreviations: F - Female; M - Male; E - Endurance; C - Control"}
#plotting enrichment analysis
e1 <- enrichment_chart(selectedClusters1, plot_by_cluster = TRUE)
e2 <- enrichment_chart(selectedClusters2, plot_by_cluster = TRUE)

#help with plot_grid: https://wilkelab.org/cowplot/articles/plot_grid.html
plot_grid(e1, e2, labels = c(comparison1, comparison2))
```

```{r, results = FALSE, fig.height = 8, fig.cap="Comparison of functional enrichment networks between female and male endurance athletes, relative to their respective controls, and considering the top ten pathways with the lowest combined p-values"}

#performing comparison between results and looking at what fields are available. Plot is set to false to not automatically plot, since the combined_results_graph function has more options.
resultComparison <- combine_pathfindR_results(result_A = combinedResult[[1]]$result, result_B = combinedResult[[2]]$result, plot_common = FALSE)
names(resultComparison)

#order by p-value so the top results with the lowest p-value are plotted below
selectedResultComparison <- selectedResultComparison[order(selectedResultComparison$combined_p), ]

#checking that p-values are in order
head(selectedResultComparison$combined_p)

#graphing combined gene-network results. Adding full descriptive terms to nodes and sizing according to p value
enrichPlot <- combined_results_graph(selectedResultComparison, use_description = TRUE, node_size = "p_val", selected_terms = selectedResultComparison$ Term_Description[1:10])

#adjusting to make more accessible with viridis color scale
enrichPlot + scale_colour_viridis_d(option = "D")
```

# Results and Discussion

# Reflection

When preparing this final project, I ran into several warning messages when trying out new packages and coding techniques. At times, it seemed like I was continuously running into these challenges and like these difficulties wouldn't go away. In general, this course has been a continual learning experience, where I never felt at the beginning of an assignment that I knew how to do it; there was always a lot of trial and error. I am someone who always seeks out challenges and I was glad that I tried to push myself in this course, but it can also be difficult to always feel like you're running into challenges (i.e., R warning messages). However, there were also a few moments where I realized that I wasn't getting the errors I had in the past, and that meant I must be able to figure out this one too. I took a break, came back, and systematically tested out possible solutions, ultimately arriving at something that worked. Looking back now after finishing this assignment, I'm pretty amazed at how much I've learned during just one term, and with many other projects going on at the same time. This is more of a philosophical point, but I do think this course has been a great testament to what happens when you do consistently challenge yourself, but to also take a few moments like these to see that these challenges are not permanent, and that they help you to improve quickly and to make tasks that were once hard feel like second nature. There were a lot of things that were very new to me when I started this course, but I've managed to learn them. In addition to the specific skills I've learned, I think a really important lesson for me has been to trust that, by calmly and methodically approaches challenges, I can learn anything new.

# Acknowledgments

No outside help was consulted in this assignment, except for cheat sheets for R (Rmarkdown, GGplot, etc.), linked tutorials, and cited vignettes.

# References

@article{Chapman2020, author = {Chapman, Mark A. and Arif, Muhammad and Emanuelsson, Eric B. and Reitzner, Stefan M. and Lindholm, Mal{'{e}}ne E. and Mardinoglu, Adil and Sundberg, Carl Johan}, doi = {10.1016/J.CELREP.2020.107808}, file = {:C:/Users/Erik/AppData/Local/Mendeley Ltd./Mendeley Desktop/Downloaded/Chapman et al. - 2020 - Skeletal Muscle Transcriptomic Comparison between Long-Term Trained and Untrained Men and Women.pdf:pdf}, issn = {2211-1247}, journal = {Cell Reports}, keywords = {RNA sequencing,endurance training,exercise physiology,gene expression,genome-scale metabolic model,human subjects,resistance training,skeletal muscle,skeletal muscle metabolism,skeletal muscle transcriptomics}, month = {jun}, number = {12}, pages = {107808}, pmid = {32579934}, publisher = {Cell Press}, title = {{Skeletal Muscle Transcriptomic Comparison between Long-Term Trained and Untrained Men and Women}}, volume = {31}, year = {2020} }

Blighe, K, S Rana, and M Lewis. 2018. "EnhancedVolcano: Publication-ready volcano plots with enhanced colouring and labeling." <https://github.com/kevinblighe/EnhancedVolcano>.

Chapman, Mark A., Muhammad Arif, Eric B. Emanuelsson, Stefan M. Reitzner, Maléne E. Lindholm, Adil Mardinoglu, and Carl Johan Sundberg. 2020. "Skeletal Muscle Transcriptomic Comparison between Long-Term Trained and Untrained Men and Women." *Cell Reports* 31 (12): 107808. <https://doi.org/10.1016/J.CELREP.2020.107808>.

Dündar, Friederike. 2020. "Performing Differential Gene Expression Analysis." \<<https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_not> es/08_practical_DE.pdf\>.

Ulgen, Ege, Ozan Ozisik, and Osman Ugur Sezerman. 2019. "PathfindR: An R Package for Comprehensive Identification of Enriched Pathways in Omics Data Through Active Subnetworks." *Frontiers in Genetics* 10 (SEP). <https://doi.org/10.3389/FGENE.2019.00858>.
