---
title: "Functional Enrichment Analysis of Sex-specific Muscle Transcriptome from Sedentary,
  Strength-Trained, or Endurance-Trained Participants"
author: "Erik Dassoff"
date: "`r Sys.Date()`"
output:
  html_document:
    df_print: paged
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, comment = "", message = FALSE)

```

#Introduction

#

# Part 1: Data Acquisition, Exploration, Filtering, and Quality Control

```{r, Part1, fig.height=15}
#attributions: This script was adapted from vignettes for enhanced volcano plots (Blighe et al. 2018) and functional enrichment analysis via the pathfindR package (Ulgen et al. 2019). Additionally, data was obtained from Chapman et al. (2020) for differential gene expression analysis. Additional R tutorials were consulted throughout, as indicated via hyperlinks located in the script.----

#loading libraries----
library(readxl)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(data.table)
#BiocManager::install('EnhancedVolcano')
library(EnhancedVolcano)
library(gridExtra)
library(pathfindR)
library(microbenchmark)
library(utils)
library(cowplot)

#setting adjustable parameters----
#p-value for statistical significance, considering p-value already adjusted for false discovery rate. 0.05 is selected to consider a level of statistical significance that limits the false discovery rate of all significant results to less than 5%.
pCutoff <- 0.05

#cutoff for fold change for volcano plots. Two is selected to highlight reasonably large differentially expressed genes that are likely to have high biological relevance.
FCcutoff <- 2

#p-value cutoff for significant enrichment. 0.05 selected as a standard level and to not be overly exclusive.
enrichThresh <- 0.05

#adjustment method for enrichment analysis (i.e., bonferroni or fdr). Using fdr to be consistent with differential expression analysis.
adjMethod <- "fdr"

#number of iterations for protein sub-network search. Start with ten and increase for greater coverage of functional pathways.
iterations <- 10

#gene set for enrichment analysis (choose from: "Kegg", "Reactome", "BioCarta", "GO-all", "Go-BP", "GO-CC", "GO-MF", "cell_markers", "mmu_KEGG"). Starting with KEGG, because it contains biological pathways of interest with good interpretability.
geneSet <- "KEGG"

#setting the protein interaction network database to search to create gene interaction networks (choose from: "Biogrid", "STRING", "KEGG", "GeneMania", "IntAct", "mmu_STRING"). Starting with KEGG to be consistent with enrichment analysis choice.
pinName <- "KEGG"

#setting two algorithms to try for search optimization and to test the execution time (choose from greedy search ["GR"], simulated annealing ["SA"], and genetic algorithm ["GA"]). Starting with greedy search, since it is performs a truncated optimization to approximate active protein interaction network, but with less computational demands, since it isn't searching until finding a true optimum, like the SA method.
searchMethod <- "GR"

#input the terms that are desired to be compared. The code currently accepts two comparisons to controls, with the intent to further compare these differential gene expression profiles.
comparison1 <- "FEvsFC"
comparison2 <- "MEvsMC"

#loading dataset----
#Muscle tissue differential gene expression data from the following article as a downloadable excel file in the xlsx format: https://www.sciencedirect.com/science/article/pii/S2211124720307890#abs0015

#insert path to file
path <- "C:/Users/Erik/OneDrive - University of Guelph/Documents/BINF/enduranceTrainingDEG.xlsx"

#the data is being read into R with the readxl package. lapply is used to load every sheet in the excel file.
DEGlist <- lapply(excel_sheets(path), read_excel, path = path)

#check
class(DEGlist)

#finding sheet names. Help: https://readxl.tidyverse.org/reference/excel_sheets.html
sheetNames <- excel_sheets(path)

#renaming sheet names in list
names(DEGlist) <- sheetNames

#check
names(DEGlist)

#combining all comparisons into one data frame to make it easier to create multiple figures in one grid using facet wrap, and for other downstream processing.
allDEGcomparisons <- rbindlist(DEGlist, idcol = "comparison")

#checking that all group comparisons are in the new data frame
unique(allDEGcomparisons$comparison)
view(allDEGcomparisons)

#adding negative log10 p-values for downstream analysis
allDEGcomparisons$negLogP <- -log10(allDEGcomparisons$`P-Adjusted`)

#checking that column was added and that values are correct
names(allDEGcomparisons)
all.equal(allDEGcomparisons$negLogP, -log10(allDEGcomparisons$`P-Adjusted`))

#selecting variables to set up format for pathfindR downstream
selectForPathfindR <- function (x) {
  x %>%
    select(gene_name, Log2FoldChange, `P-Adjusted`)
}

DEGlistPathfindR <- lapply(DEGlist, selectForPathfindR)

#check that column names are as expected
lapply(DEGlistPathfindR, function(x) {names(x)})


#exploring data----

#creating function to explore data in every element of the list
explore <- function(x) {
  class = class(x)
  names = names(x)
  dimensions = dim(x)
  return(list(class = class, names = names, dimensions = dimensions))
}

lapply(DEGlist, explore)

#creating function to better understand the data values
dataStats <- function(x) {
  pValStats = summary(x$'P-Adjusted')
  FCstats = summary(x$'Log2FoldChange')
  numGeneNames = length(unique(x$'gene_name'))
  directions = table(x$'Direction')
  return(list("P Value Stats" = pValStats, "Fold Change Stats" = FCstats, "Number of Unique Genes" = numGeneNames, "Change Directions" = directions))
}

lapply(DEGlist, dataStats)

#checking to see if there are the same number of unique gene names as gene names overall for each list element. If so, these can serve as the unique identifiers for each element in each list.
lapply(DEGlist, function(x) {length(x$gene_name) == length(unique(x$gene_name))})

```

```{r, Quality, fig.height=4, fig.cap="Amount of significantly differentially regulated genes by p-value and group comparison. Abbreviations: F - Female; M - Male; E - Endurance; C - Control"}
#advice for quality checking for differential gene expression analysis: https://physiology.med.cornell.edu/faculty/skrabanek/lab/angsd/lecture_notes/08_practical_DE.pdf

#help with facetwrap:http://zevross.com/blog/2019/04/02/easy-multi-panel-plots-in-r-using-facet_wrap-and-facet_grid-from-ggplot2/

#plotting histogram of p-values for all comparisons with color to identify which are up- or down-regulated. Any comparisons with only up- or down-regulated genes are likely to not be of high quality.
GGhist <- ggplot(allDEGcomparisons, aes(`P-Adjusted`, fill = Direction))
GGhist +
  geom_histogram(binwidth = 0.001, col = I("grey10")) +
  labs(y = "Number of Differentially Expressed Genes", x = "Adjusted P-Value") +
  theme_classic() +
  facet_wrap(~allDEGcomparisons$comparison) +
  scale_fill_viridis_d(option = "D")

```

```{r, Volcano, fig.height=12, fig.cap="Volcano plots of differentially expressed genes for all combinations of male and female endurance athletes and their controls. Abbreviations: F - Female; M - Male; E - Endurance; C - Control; P - Significant (adj. P < 0.05) differential expression; FC_P - Significant differential expression (adj. P < 0.05) with fold change > 2", warning = FALSE}
#creating volcano plots to visualize significantly differentially expressed genes with high significance

#enhanced volcano plot vignette: https://bioconductor.org/packages/devel/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html

#creating a function for the volcano plot, to visualize significantly differentially expressed genes with high significance. Formatting is adjusted to fit in the Rmarkdown output.
volcanoPlot <- function(data, title) {EnhancedVolcano(data,
                lab = data$gene_name,
                x = 'Log2FoldChange',
                y = 'P-Adjusted',
                title = title,
                titleLabSize = 14,
                legendLabSize = 14,
                legendIconSize = 6,
                axisLabSize = 14,
                legendPosition = "top",
                pCutoff = pCutoff,
                FCcutoff = FCcutoff,
                pointSize = 3,
                labSize = 4,
                boxedLabels = TRUE,
                drawConnectors = TRUE,
                gridlines.minor = FALSE)}

#plotting result for each comparison.
p1 <- volcanoPlot(DEGlist$FEvsFC, "FE vs FC")
p2 <- volcanoPlot(DEGlist$MEvsMC, "ME vs MC")
p3 <- volcanoPlot(DEGlist$MCvsFC, "MC vs FC")
p4 <- volcanoPlot(DEGlist$MSvsME, "MS vs ME")
p5 <- volcanoPlot(DEGlist$MEvsFE, "ME vs FE")
p6 <- volcanoPlot(DEGlist$MSvsMC, "MS vs MC")

#change colours to viridis scale
viridis <- function(p) {
  p + scale_colour_viridis_d(option = "D")
}

p1 <- viridis(p1)
p2 <- viridis(p2)
p5 <- viridis(p5)
p3 <- viridis(p3)

#arranging selected figures into a grid
grid.arrange(p1, p2, p5, p3, ncol = 2)
```

#Part 2:

```{r, Enrichment}
#create functional annotation ---- using pathfindR: https://pubmed.ncbi.nlm.nih.gov/31608109/

#full tutorial: https://cran.r-project.org/web/packages/pathfindR/vignettes/intro_vignette.html

#converting to data frame for all comparisons, which is an essential class structure to be able to run downstream pathfindR analysis
DEGlistPathfindR <- lapply(DEGlistPathfindR, function (x) {as.data.frame(x)})

#check
lapply(DEGlistPathfindR, function (x) {class(x)})

#turning off scientific notation to match pathway analysis format
options(scipen = 999)

#check that values are not in scientific notation
summary(DEGlistPathfindR$FEvsFC$`P-Adjusted`)

#creating function for running pathfindR, which will be used more than once downstream. Not plotting the enrichment chart, so that it can be plotted separately later.
pathfindR <- function(data) {run_pathfindR(data,
  adj_method = adjMethod,
  gene_sets = geneSet,
  pin_name_path = pinName,
  search_method = "GS",
  enrichment_threshold = enrichThresh,
  plot_enrichment_chart = FALSE)}

#creating custom function for selecting key outputs of interest: clustering pathway enrichment with agglomerative hierarchical clustering and plotting the enriched terms as well as the gene interaction networks for each comparison.
enrichAnalysis <- function(data) {
  result <- run_pathfindR(data)
  clusteredEnrichment <- cluster_enriched_terms(result, plot_clusters_graph = FALSE)
  termGeneGraph <- term_gene_graph(result, use_description = TRUE)
  return(list("result" = result, "enrichment" = clusteredEnrichment, "termPlot" = termGeneGraph))
}

#creating pre-specified search list
df1 <- DEGlistPathfindR[[comparison1]]
df2 <- DEGlistPathfindR[[comparison2]]
inputList <- list(df1, df2)
remove(df1, df2)

#checking that the list contains two data frames
class(inputList)
length(inputList)
class(inputList[[1]])
view(inputList[[1]])

#running analysis function through pre-selected comparisons
combinedResult <- lapply(inputList, enrichAnalysis)

#viewing tabular results
view(combinedResult[[1]]$result)
view(combinedResult[[1]]$enrichment)

#creating a subset of samples with fold enrichment >2, low p-values, and only the representative terms for each cluster, to simplify figure by not having several terms within each cluster.
selectedClusters1 <- subset(combinedResult[[1]]$enrichment, Status %in% "Representative" & Fold_Enrichment > 1.5 & lowest_p < 0.01)

selectedClusters2 <- subset(combinedResult[[2]]$enrichment, Status %in% "Representative" & Fold_Enrichment > 1.5 & lowest_p <0.01)

#checking that subsetting was performed
length(combinedResult[[1]]$enrichment$ID) > length(selectedClusters1$ID)
```

```{r, results = FALSE, fig.height = 10, fig.cap="Functional enrichment analysis for female and male endurance athletes compared to their respective sedentary controls. All terms with enrichment p-values < 0.01 and fold change values above 1.5 are included. Abbreviations: F - Female; M - Male; E - Endurance; C - Control"}
#plotting enrichment analysis
e1 <- enrichment_chart(selectedClusters1, plot_by_cluster = TRUE)
e2 <- enrichment_chart(selectedClusters2, plot_by_cluster = TRUE)

#help with plot_grid: https://wilkelab.org/cowplot/articles/plot_grid.html
plot_grid(e1, e2, labels = c(comparison1, comparison2))
```

```{r, results = FALSE, fig.height = 8, fig.cap="Comparison of functional enrichment networks between female and male endurance athletes, relative to their respective controls, and considering the top ten pathways with the lowest combined p-values"}

#performing comparison between results and looking at what fields are available. Plot is set to false to not automatically plot, since the combined_results_graph function has more options.
resultComparison <- combine_pathfindR_results(result_A = combinedResult[[1]]$result, result_B = combinedResult[[2]]$result, plot_common = FALSE)
names(resultComparison)

#order by p-value so the top results with the lowest p-value are plotted below
selectedResultComparison <- selectedResultComparison[order(selectedResultComparison$combined_p), ]

#checking that p-values are in order
head(selectedResultComparison$combined_p)

#graphing combined gene-network results. Adding full descriptive terms to nodes and sizing according to p value
enrichPlot <- combined_results_graph(selectedResultComparison, use_description = TRUE, node_size = "p_val", selected_terms = selectedResultComparison$ Term_Description[1:10])

#adjusting to make more accessible with viridis color scale
enrichPlot + scale_colour_viridis_d(option = "D")
```

#References
